internal/ssm/install.go

```diff
@@ -18,37 +21,104 @@ import (
 	"github.com/aws/eks-hybrid/internal/util/cmd"
 )
 
-const (
-	installerPath = "/opt/ssm/ssm-setup-cli"
-	configRoot    = "/etc/amazon"
+// TODO: a better way to override this
+var (
+	defaultInstallerPath = "/opt/ssm/ssm-setup-cli"
+	configRoot           = "/etc/amazon"
 )
 
 // Source serves an SSM installer binary for the target platform.
 type Source interface {
 	GetSSMInstaller(ctx context.Context) (io.ReadCloser, error)
+	GetSSMInstallerSignature(ctx context.Context) (io.ReadCloser, error)
+	PublicKey() string
 }
 
 // PkgSource serves and defines the package for target platform
 type PkgSource interface {
 	GetSSMPackage() artifact.Package
 }
 
-func Install(ctx context.Context, tracker *tracker.Tracker, source Source) error {
+func InstallTo(ctx context.Context, tracker *tracker.Tracker, source Source, logger *zap.Logger, installerPath string) error {
```

Comment:
we should prob think to group the last 4 in a struct, 5 arguments is starting to be too high